### PR TITLE
Add libxmtp version to requests

### PR DESCRIPTION
**Pull request by neekolas**
## tl;dr

- Starts setting the `x-app-version` and `x-libxmtp-version` headers in all requests

## Issues
- https://github.com/xmtp/libxmtp/issues/967

## More info

- The `libxmtp_version` gets taken from the `version` field in `xmtp_mls/Cargo.toml`. This gets set from the `[workspace.package]` version in the root `Cargo.toml` which is inherited by everything else

### CODE REVIEW

**Comment by neekolas** [comment]
* **#1011** <a href="https://app.graphite.dev/github/pr/xmtp/libxmtp/1011?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @neekolas and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: xmtp_api_http/src/lib.rs [hunk 1]
```diff
@@ -50,6 +55,43 @@ impl XmtpHttpApiClient {
...
+impl ClientWithMetadata for XmtpHttpApiClient {
+    fn set_app_version(&mut self, version: String) -> Result<(), Error> {
+        self.app_version = Some(version);
+
+        let mut headers = header::HeaderMap::new();
+        if let Some(app_version) = &self.app_version {
+            headers.insert("x-app-version", app_version.parse().unwrap());
+        }
+        if let Some(libxmtp_version) = &self.libxmtp_version {
+            headers.insert("x-libxmtp-version", libxmtp_version.parse().unwrap());
```
Comment (insipx) [inline xmtp_api_http/src/lib.rs:67]: probalby don't want to be unwrapping -- could default or error

File: mls_validation_service/Cargo.toml [hunk 2]
```diff
@@ -1,7 +1,7 @@
 [package]
 edition = "2021"
 name = "mls_validation_service"
-version = "0.1.0"
+version = "0.0.1"
```
[the remark this answers is not in the document]
Comment (insipx) [reply mls_validation_service/Cargo.toml]: what's wrong with `workspace.version`?

[the remark this answers is not in the document]
Comment (insipx) [reply mls_validation_service/Cargo.toml]: Ah for the validation service

I think we can still version the package separately, by not giving it the `version.workspace`

File: xmtp_api_http/src/lib.rs (hunk 1 above)
Comment (neekolas) [reply xmtp_api_http/src/lib.rs:67]: Ya. That's a good call out

**Comment by neekolas** [comment]
@insipx 
> what's wrong with version.workspace?

Think you got there just before I pushed up a change to do that everywhere I could

**Review by insipx — APPROVED** [review]

**Pull request merged**